### ISSUE TITLE
DoesNotExist: File matching query does not exist.

### ISSUE THREAD

**Issue by HazAT** [getsentry/sentry#8778]
https://sentry.io/sentry/sentry/issues/528067078/
https://sentry.io/sentry/sentry/issues/529591799/events/22831434706/

```
DoesNotExist: File matching query does not exist.
(4 additional frame(s) were not displayed)
...
  File "sentry/api/endpoints/dsym_files.py", line 201, in post
    'associatedDsymFiles': serialize(associated, request.user),
  File "sentry/api/serializers/base.py", line 42, in serialize
    return [serializer(o, attrs=attrs.get(o, {}), user=user, *args, **kwargs) for o in objects]
  File "sentry/api/serializers/base.py", line 57, in __call__
    return self.serialize(obj, attrs, user, *args, **kwargs)
  File "sentry/api/serializers/models/dsym_file.py", line 18, in serialize
    'symbolType': obj.dsym_type,
  File "sentry/models/dsymfile.py", line 235, in dsym_type
    ct = self.file.headers.get('Content-Type', 'unknown').lower()

DoesNotExist: File matching query does not exist.
```

**Comment by HazAT** [comment]
Sorry @jan-auer for adding you here, I would like to sort this out since more and more people are hitting this kind of error.

**Comment by pyrou** [comment]
Any update ? I'm also experiencing this issue with Sentry onpremise docker 9.0.0 when I try to upload dSYMs with official fastlane plugin

**Comment by jan-auer** [comment]
Several causes for this issue have been fixed in #9549, #9947 and #9974, and will be available with the next release of sentry. At the moment I don't have a timeline for the next release, but you can always build and deploy the latest `master`.

We're currently working on another feature that will change the way we handle debug files internally, see #10052.
